action/protocol/staking/handler_candidate_selfstake.go

```diff
@@ -0,0 +1,94 @@
+package staking
+
+import (
+	"context"
+	"math"
+	"math/big"
+
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+)
+
+const (
+	handleCandidateSelfStake = "candidateSelfStake"
+
+	candidateNoSelfStakeBucketIndex = math.MaxUint64
+)
+
+func (p *Protocol) handleCandidateSelfStake(ctx context.Context, act *action.CandidateActivate, csm CandidateStateManager,
+) (*receiptLog, []*action.TransactionLog, error) {
+	actCtx := protocol.MustGetActionCtx(ctx)
+	featureCtx := protocol.MustGetFeatureCtx(ctx)
+	log := newReceiptLog(p.addr.String(), handleCandidateSelfStake, featureCtx.NewStakingReceiptFormat)
+
+	// caller must be the owner of a candidate
+	cand := csm.GetByOwner(actCtx.Caller)
+	if cand == nil {
+		return log, nil, errCandNotExist
+	}
+
+	bucket, rErr := p.fetchBucket(csm, act.BucketID())
+	if rErr != nil {
+		return log, nil, rErr
+	}
+
+	if err := p.validateBucketSelfStake(ctx, csm, NewEndorsementStateManager(csm.SM()), bucket, cand); err != nil {
+		return log, nil, err
+	}
```

Comment:
`log.AddTopics(byteutil.Uint64ToBytesBigEndian(bucket.Index), bucket.Candidate.Bytes())`

Reply:
Indeed, we need add log to indicate what we do in this action